tests/php/src/BlockUniqidTransformerTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace AmpProject\AmpWP\Tests;
+
+use AMP_Block_Uniqid_Sanitizer;
+use AMP_Options_Manager;
+use AMP_Theme_Support;
+use AmpProject\AmpWP\BlockUniqidTransformer;
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AmpProject\AmpWP\Option;
+use AmpProject\AmpWP\Tests\Helpers\MarkupComparison;
+
+/** @coversDefaultClass \AmpProject\AmpWP\BlockUniqidTransformer */
+final class BlockUniqidTransformerTest extends TestCase {
+
+	use MarkupComparison;
+
+	/** @var BlockUniqidTransformer */
+	private $instance;
+
+	/**
+	 * Setup.
+	 *
+	 * @inheritdoc
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->instance = new BlockUniqidTransformer();
+	}
+
+	public function test_it_can_be_initialized() {
+		$this->assertInstanceOf( Conditional::class, $this->instance );
+		$this->assertInstanceOf( Registerable::class, $this->instance );
+		$this->assertInstanceOf( Service::class, $this->instance );
+	}
+
+	/**
+	 * @covers ::is_needed()
+	 * @covers ::has_gutenberg_plugin()
+	 */
+	public function test_is_needed() {
+		if (
+			! defined( 'GUTENBERG_VERSION' )
```

Comment:
I think this needs to be changed to check if `GUTENBERG_VERSION` is between the versions that used `uniqid()`.